File: backend/springnatour/src/main/java/com/unina/springnatour/dto/report/ReportDto.java
package com.unina.springnatour.dto.report;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.unina.springnatour.dto.route.RouteTitleDto;
import com.unina.springnatour.dto.user.UserDto;
import com.unina.springnatour.model.User;
import com.unina.springnatour.model.route.Route;
import lombok.Data;

import java.io.Serializable;

@Data
public class ReportDto implements Serializable {
    @JsonProperty("report_id")
    private Long reportId;

    @JsonProperty("report_title")
    private String reportTitle;

    @JsonProperty("report_description")
    private String reportDescription;

    @JsonProperty("author")
    private UserDto author;

    @JsonProperty("reported_route")
    private RouteTitleDto reportedRoute;
}

File: client/app/src/main/java/com/unina/natourkt/core/data/remote/dto/ReportDto.kt
package com.unina.natourkt.core.data.remote.dto

import com.google.gson.annotations.SerializedName

data class ReportDto(
    @SerializedName("report_id")
    val reportId: Long,

    @SerializedName("report_title")
    val reportTitle: String,

    @SerializedName("report_description")
    val reportDescription: String,

    @SerializedName("author")
    val author: UserDto,

    @SerializedName("reported_route")
    val reportedRoute: RouteTitleDto
)